core/common/src/main/java/alluxio/network/protocol/databuffer/DataByteBuffer.java

```diff
@@ -52,6 +54,8 @@ public ByteBuffer getReadOnlyByteBuffer() {
 
   @Override
   public void release() {
-    // Nothing we need to release explicitly, let GC take care of all objects.
+    if (mBuffer.isDirect()) {
+      BufferUtils.cleanDirectBuffer(mBuffer);
```

Comment:
Yeah, at least we should not call release if we are sharing the buffer (we would need to have some kind of ref count before calling release).